src/base/bittorrent/session.cpp

```diff
@@ -3123,7 +3124,7 @@ void Session::handleIPFilterError()
 }
 
 #if LIBTORRENT_VERSION_NUM < 10100
-void Session::dispatchAlerts(std::auto_ptr<libt::alert> alertPtr)
+void Session::dispatchAlerts(std::unique_ptr<libt::alert> &alertPtr)
```

Comment:
Hmm, shouldn't plain old data pointer fit as? Kind of 
```C++
void Session::dispatchAlerts(libtorrent::alert *alertPtr)
```
and 
```C++
dispatchAlerts(alertPtr.release());
```
The only downside to rvalue / move magic is possibility of memory leaking (size of lost alert) in case of exception happening. By if we got exception here, it will be memory allocation exception and app will crush later anyway.